pkg/cloud/awsprovider.go

```diff
@@ -643,17 +643,27 @@ func (k *awsKey) Features() string {
 	region, _ := util.GetRegion(k.Labels)
 
 	key := region + "," + instanceType + "," + operatingSystem
-	usageType := PreemptibleType
+	usageType := k.getUsageType(k.Labels)
 	spotKey := key + "," + usageType
 	if l, ok := k.Labels["lifecycle"]; ok && l == "EC2Spot" {
 		return spotKey
 	}
 	if l, ok := k.Labels[k.SpotLabelName]; ok && l == k.SpotLabelValue {
 		return spotKey
 	}
```

Comment:
@AjayTripathy what's the intention of `k.SpotLabelName` and `k.SpotLabelValue`? When I run locally targeting an EKS cluster, these values are nil and can never enter this. The lines below work, but feels a bit hacky to me.

Reply:
I think this was to support overriding in non EKS implementations of k8s on AWS like kops. Can safely be ignored on EKS, I would think.

Reply:
Got it, thank you! I added an additional check to see if the `usageType` returned is the same as `PreemptibleType` and return the `spotKey` if it is. I'm open to a different way of checking this.